### PR TITLE
feat(api-service,dashboard): create/update layout validation fixes NV-6165

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?

Layouts:
- improved create layout functionality that will now generate the default control values for block Maily editor with `content` variable
- create/update layout validation with issues generation
- refactored the common logic to generate issues

### Screenshots

https://github.com/user-attachments/assets/14024035-752e-4b2b-af18-3779d47ea147

## Summary by CodeRabbit

* **New Features**
  * Introduced comprehensive runtime issue validation and reporting for layouts and workflow steps, including schema and Liquid template validation, with unified error handling and feedback in the UI.
  * Added default layout templates for email content initialization.

* **Bug Fixes**
  * Improved error handling and user feedback for layout creation and update failures, providing detailed validation messages.

* **Refactor**
  * Unified and simplified issue types across backend and frontend by introducing a new `RuntimeIssue` model, replacing previous workflow-specific issue types and enums.
  * Updated API responses and UI components to use the new issue structure, ensuring consistency and easier integration.
  * Removed deprecated and redundant code related to old issue enums and interfaces.

* **Tests**
  * Added extensive tests for layout upsert functionality, including validation, error handling, and edge cases.

* **Chores**
  * Updated import paths and reorganized utility functions for improved maintainability.

### CODE REVIEW

**Comment by LetItRock** [comment]
@coderabbitai summary

File: apps/api/src/app/layouts-v2/dtos/layout-response.dto.ts [hunk 1]
```diff
@@ -63,17 +61,6 @@ export class LayoutResponseDto {
...
-  @ApiPropertyOptional({
-    description: 'Runtime issues for layout creation and update',
-    type: 'object',
-    additionalProperties: {
-      $ref: getSchemaPath(RuntimeIssueDto),
-    },
-  })
-  @IsOptional()
-  @Type(() => RuntimeIssueDto)
-  issues?: Record<LayoutCreateAndUpdateKeys, RuntimeIssueDto>;
```
Comment (LetItRock) [inline apps/api/src/app/layouts-v2/dtos/layout-response.dto.ts:75]: issues are not part of the layout response dto, but generated as HTTP 400 error

File: apps/api/src/app/layouts-v2/layouts.controller.ts [hunk 2]
```diff
@@ -75,6 +76,12 @@ export class LayoutsController {
       UpsertLayoutCommand.create({
         layoutDto: {
           ...createLayoutDto,
+          controlValues: {
+            email: {
+              body: JSON.stringify(EMPTY_LAYOUT),
+              editorType: 'block',
+            },
+          },
```
Comment (LetItRock) [inline apps/api/src/app/layouts-v2/layouts.controller.ts:84]: when the layout is created these are the control value defaults

File: apps/api/src/app/layouts-v2/usecases/build-layout-issues/build-layout-issues.usecase.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    ) {
+      contentIssues.controls = {
+        'email.body': [
+          {
+            message: `The layout body should contain the "${LAYOUT_CONTENT_VARIABLE}" variable`,
+            issueType: ContentIssueEnum.MISSING_VALUE,
+          },
+        ],
+      };
+    }
```
Comment (LetItRock) [inline apps/api/src/app/layouts-v2/usecases/build-layout-issues/build-layout-issues.usecase.ts:62]: check if `content` variable was used

File: apps/api/src/app/layouts-v2/usecases/build-layout-issues/build-layout-issues.usecase.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      controlValues: sanitizedControlValues ?? {},
+    });
+
+    const liquidIssues: ControlIssues = {};
+    processControlValuesByLiquid({
+      variableSchema: layoutVariablesSchema,
+      currentValue: controlValues ?? {},
+      currentPath: [],
+      issues: liquidIssues,
+    });
```
Comment (LetItRock) [inline apps/api/src/app/layouts-v2/usecases/build-layout-issues/build-layout-issues.usecase.ts]: these functions were extracted from the step issues build usecase and reused here

File: apps/api/src/app/layouts-v2/usecases/upsert-layout/upsert-layout.usecase.ts [hunk 5]
```diff
@@ -137,22 +151,32 @@ export class UpsertLayoutUseCase {
...
+      BuildLayoutIssuesCommand.create({
+        controlSchema: layoutControlSchema,
+        controlValues,
+        resourceOrigin: command.layoutDto.__source ? ResourceOriginEnum.NOVU_CLOUD : ResourceOriginEnum.EXTERNAL,
+        user: command.user,
+      })
+    );
+
+    if (Object.keys(issues).length > 0) {
+      throw new BadRequestException(issues);
```
Comment (LetItRock) [inline apps/api/src/app/layouts-v2/usecases/upsert-layout/upsert-layout.usecase.ts:170]: build issues, and if any keys then throw bad request

File: apps/api/src/app/layouts-v2/utils/layout-templates.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        {
+          type: 'variable',
+          attrs: {
+            id: 'content',
+            label: null,
+            fallback: null,
+            required: false,
+            aliasFor: null,
+          },
+        },
```
Comment (LetItRock) [inline apps/api/src/app/layouts-v2/utils/layout-templates.ts:17]: the empty layout on creation will only have the `content` variable

File: apps/api/src/app/layouts-v2/utils/layout-templates.ts [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+      ],
+    },
+  ],
+};
+
+export const DEFAULT_LAYOUT = {
```
Comment (LetItRock) [inline apps/api/src/app/layouts-v2/utils/layout-templates.ts:23]: the default layout will be used in later PRs

File: apps/api/src/app/workflows-v2/dtos/base-step-issue.dto.ts [hunk 8]
```diff
@@ -1,14 +1,14 @@
 import { ApiProperty } from '@nestjs/swagger';
 import { IsEnum } from 'class-validator';
-import { StepIssueEnum } from '@novu/shared';
+import { ContentIssueEnum, IntegrationIssueEnum } from '@novu/shared';
```
Comment (LetItRock) [inline apps/api/src/app/workflows-v2/dtos/base-step-issue.dto.ts:3]: step content and integration issue enums were renamed

File: apps/api/src/app/workflows-v2/dtos/step-issues.dto.ts [hunk 9]
```diff
@@ -17,7 +18,7 @@ export class StepIssuesDto {
   })
   @IsOptional()
   @ValidateNested()
-  controls?: Record<string, StepContentIssueDto[]>;
+  controls?: Record<string, RuntimeIssue[]>;
```
Comment (LetItRock) [inline apps/api/src/app/workflows-v2/dtos/step-issues.dto.ts:21]: created the `RuntimeIssue` class that is reused for both steps and layouts

File: apps/dashboard/src/components/layouts/layout-editor-provider.tsx [hunk 10]
```diff
@@ -78,11 +80,49 @@ export const LayoutEditorProvider = ({ children }: { children: React.ReactNode }
...
+        setFormIssues(controlIssues);
+
+        const firstControlError = getFirstErrorMessage({ controls: controlIssues }, 'controls');
+        showErrorToast(
+          firstControlError?.message ?? 'Failed to update layout',
+          'Failed to update layout',
+          toastOptions
+        );
+        return;
+      }
```
Comment (LetItRock) [inline apps/dashboard/src/components/layouts/layout-editor-provider.tsx:124]: when got bad request then set the form issues and show the toast with the first error

File: apps/api/src/app/workflows-v2/dtos/step-issues.dto.ts (hunk 9 above)
Comment (scopsy) [reply apps/api/src/app/workflows-v2/dtos/step-issues.dto.ts:21]: Awesome!

**Review by scopsy — APPROVED** [review]
